packages/core/test/merkle-distributor/MerkleDistributor.js

```diff
@@ -315,166 +322,350 @@ contract("MerkleDistributor.js", function(accounts) {
+            amount: leaf2.amount,
+            merkleProof: merkleTrees[1].getProof(leaf2.leaf)
+          }
+        ];
+        await merkleDistributor.claimMulti(claims, rewardToken.address, leaf1.account, { from: rando });
+
+        // Account 0 should have gained claimed amount from both leaves.
+        const batchedClaimAmount = toBN(leaf1.amount).add(toBN(leaf2.amount));
+        assert.equal(
+          (await rewardToken.balanceOf(leaf1.account)).toString(),
+          accountBalanceBefore.add(batchedClaimAmount).toString()
+        );
 
-      const accountBalanceBefore = await rewardToken.balanceOf(leaf1.account);
+        // One Claimed event should have been emitted for each batched claim.
+        const claimedEvents = await merkleDistributor.getPastEvents("Claimed");
+        assert.equal(claimedEvents.length, claims.length);
+      });
+      it("One tree: Can only batch claim for one account", async function() {
+        // Leaf 2 is for account[1], can't batch claim for two different accounts.
+        const leaf1 = rewardLeafs[0][0];
+        const leaf2 = rewardLeafs[0][1];
 
-      const claims = [
-        {
-          windowIndex: windowIndex,
-          account: leaf1.account,
-          amount: leaf1.amount,
-          merkleProof: merkleTree1.getProof(leaf1.leaf)
-        },
-        {
-          windowIndex: windowIndex + 1,
-          account: leaf2.account,
-          amount: leaf2.amount,
-          merkleProof: merkleTree2.getProof(leaf2.leaf)
+        const invalidClaims = [
+          {
+            windowIndex: windowIndex,
+            account: leaf1.account,
+            accountIndex: leaf1.accountIndex,
+            amount: leaf1.amount,
+            merkleProof: merkleTrees[0].getProof(leaf1.leaf)
+          },
+          {
+            windowIndex: windowIndex,
+            account: leaf2.account,
+            accountIndex: leaf2.accountIndex,
+            amount: leaf2.amount,
+            merkleProof: merkleTrees[0].getProof(leaf2.leaf)
+          }
+        ];
+
+        // Show that verifyProof works for claims individually.
+        for (let claim of invalidClaims) {
+          assert.isTrue(await merkleDistributor.verifyClaim(claim));
         }
-      ];
-      const claimTx = await merkleDistributor.claimWindows(claims, rewardToken.address, leaf1.account, { from: rando });
-      assert.equal(claimTx.receipt.gasUsed, 131228);
 
-      // Account 0 should have gained claimed amount from both leaves.
-      const batchedClaimAmount = toBN(leaf1.amount).add(toBN(leaf2.amount));
-      assert.equal(
-        (await rewardToken.balanceOf(leaf1.account)).toString(),
-        accountBalanceBefore.add(batchedClaimAmount).toString()
-      );
+        // But batching them together fails.
+        assert(await didContractThrow(merkleDistributor.claimMulti(invalidClaims, rewardToken.address, leaf1.account)));
+      });
+      it("Two trees: Can only batch claim for one reward currency", async function() {
```

Comment:
@daywiss new test based on the bug you found, showing that its now impossible to make batch claim for multiple reward currencies.